js/ad_manager_ooyala.js

```diff
@@ -0,0 +1,207 @@
+  /**
+   * @class OoyalaAdManager
+   * @classDesc The Ooyala Ads Manager class, registered as an ads manager with the ad manager controller.
+   * Controls how Ooyala videos loaded and played as ads while communicating with the ad manager framework.
+   * @public
+   * @property {string} name Name of the OoyalaAdManager, must match what is sent from Backlot and used at the page level
+   * @property {object} _amc An internal reference to the Ad Manager Fratmework
+   * @property {string} _adEmbedCode Keeps track of the embed code of the ooyala ad that is currently playing
+   * @property {string} __movieEmbedCode Keeps track of the embed code of the main video that is currently playing
+   * @property {boolean} _getMainVideoEmbedCode To get the embed code of the main video from the metadata
+   * @property {boolean} _playedAtleastOnce To keep track of the main video play
+   * @property {object} _adProperties Stores the ad data of the ad that is currently being loaded
+   */
+  var OoyalaAdManager = function() {
+    this.name = "ooyala-ads-manager";
+    this.ready = false;
+    this.movieMetadata = {};
+    this.streams = {};
+    var _amc  = null;
+    var _adEmbedCode = "";
+    var _movieEmbedCode ="";
+    var _getMainVideoEmbedCode =true;
+    var _playedAtleastOnce = false;
+    var _adProperties = {};
+    var _urlOpened = false;
+    var _adEnded = false;
+    
+    /**
+     * Called by the Ad Manager Controller. Use this function to initialize, create listeners, and load
+     * remote JS files.
+     * @method AdManager#initialize
+     * @public
+     * @param {object} adManagerController A reference to the Ad Manager Controller
+     * @param {string} playerId The unique player identifier of the player initializing the class
+     */
+    this.initialize = function(adManagerController, playerId) {
+      _amc = adManagerController;
+      if (_amc) {
+        _amc.addPlayerListener(_amc.EVENTS.INITIAL_PLAY_REQUESTED, _onInitialPlayRequested);
+        _amc.addPlayerListener(_amc.EVENTS.CONTENT_COMPLETED, _onContentCompleted);
+      }
+    };
+
+    /**
+     * Called by Ad Manager Controller. When this function is called, all movie and server metadata are
+     * ready to be parsed.
+     * This metadata may contain the adTagUrl and other ad manager and movie specific configuration.
+     * @method AdManager#loadMetadata
+     * @public
+     * @param {object} adManagerMetadata Ad manager-specific metadata
+     * @param {object} backlotBaseMetadata Base metadata from Ooyala Backlot
+     * @param {object} movieMetadata Metadata for the main video
+     */
+    this.loadMetadata = function(adManagerMetadata, backlotBaseMetadata, movieMetadata) {
+      if (!movieMetadata) return;
+      this.movieMetadata = movieMetadata;
+      if (_getMainVideoEmbedCode == true) {
+        _movieEmbedCode = movieMetadata.embed_code;
+        _getMainVideoEmbedCode = false;
+      }
+      _adEmbedCode = movieMetadata.ads[0].ad_embed_code;
```

Comment:
There needs to be a null check for movieMetadata.ads[0] existing. This will throw an error if there there are no ads for this plugin.